src/PIL/Image.py

```diff
@@ -3100,7 +3100,8 @@ def fromarray(obj, mode=None):
         try:
             mode, rawmode = _fromarray_typemap[typekey]
         except KeyError as e:
-            msg = "Cannot handle this data type: %s, %s" % typekey  # noqa: UP031
+            typ_shape, typ_type = typekey
+            msg = f"Cannot handle this data type: {typ_shape}, {typ_type}"
```

Comment:
```suggestion
            typekey_shape, typestr = typekey
            msg = f"Cannot handle this data type: {typekey_shape}, {typestr}"
```
`typestr` seems more appropriate to me, given https://github.com/python-pillow/Pillow/blob/25cc5afbb115db5449bb18a45431e94ebf5ad01c/src/PIL/Image.py#L3096